modules/tensor_mechanics/src/actions/TensorMechanicsAction.C

```diff
@@ -827,6 +790,61 @@ TensorMechanicsAction::actGatherActionParameters()
   }
 }
 
+void
+TensorMechanicsAction::actAddVariable()
+{
+  // Add displacement variables (optional)
+  if (_is_primary_action && _current_task == "add_variable")
+  {
+    auto params = _factory.getValidParams("MooseVariable");
+    // determine necessary order
+    const bool second = _problem->mesh().hasSecondOrderElements();
+
+    params.set<MooseEnum>("order") = second ? "SECOND" : "FIRST";
+    params.set<MooseEnum>("family") = "LAGRANGE";
+    if (isParamValid("scaling"))
+      params.set<std::vector<Real>>("scaling") = {getParam<Real>("scaling")};
+
+    // generate `block` parameter
```

Comment:
is this changing the name of the displacement variable to have the block name in it?  Its not in the old code.